app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -439,43 +444,60 @@ MinionPoller = {
     var isWorker = MinionPoller.selectedNodes.indexOf(minion.id) !== -1;
     var isUnused = !isMaster && !isWorker;
 
-    var masterHtml;
     var masterChecked = '';
-    var minionHtml;
     var minionChecked = '';
+    var disabledHtml = '';
+
+    if (State.pendingRemovalMinionId != null){
+      disabledHtml = 'disabled="disabled"';
+    }
 
     if (isMaster) {
       masterChecked = 'checked';
     }
 
-    masterHtml = '\
-      <input name="roles[master][]" id="roles_master_' + minion.id +
+    var masterHtml = '\
+      <input hidden name="roles[master][]" id="roles_master_' + minion.id +
       '" value="' + minion.id + '" type="checkbox" title="Select node as master" ' + masterChecked + '>';
 
     if (isWorker) {
       minionChecked = 'checked';
     }
 
-    minionHtml = '<input name="roles[worker][]" id="roles_worker_' + minion.id +
+    var minionHtml = '\
+      <input hidden name="roles[worker][]" id="roles_worker_' + minion.id +
       '" value="' + minion.id + '" type="checkbox" title="Select node for bootstrapping" ' + minionChecked + '>';
-
     var roleHtml = '\
-      <td class="role-column">' +
+      <td class="role-column" id="role-column-'+ minion.minion_id +'">' +
         masterHtml +
         minionHtml +
         '<div class="btn-group role-btn-group">\
-          <button type="button" class="btn btn-default master-btn '+ (isMaster ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '" data-minion-role="master">Master</button>\
-          <button type="button" class="btn btn-default worker-btn ' + (isWorker ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '" data-minion-role="worker">Worker</button>\
-          <button type="button" class="btn btn-default unused-btn ' + (isUnused ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '">Unused</button>\
+          <button type="button" '+ disabledHtml +' class="btn btn-default master-btn '+ (isMaster ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '" data-minion-role="master">Master</button>\
+          <button type="button" '+ disabledHtml +' class="btn btn-default worker-btn ' + (isWorker ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '" data-minion-role="worker">Worker</button>\
+          <button type="button" '+ disabledHtml +' class="btn btn-default unused-btn ' + (isUnused ? 'btn-primary' : '') + '" data-minion-id="' + minion.id + '">Unused</button>\
         </div>\
       </td>\
     ';
 
+    var actionHtml = '\
+      <td class="action-column">\
+        <a class="remove-minion"  href="#" data-minion-id="' + minion.minion_id + '" data-minion-fqdn="' + minion.fqdn + '">Remove</a>\
```

Comment:
We forgot to check if there's a pending removal to render this already disabled. Just need to add the class `disabled` if there's a pending removal.

Reply:
Thank you (Y)